# Superheroes
- Andrea Cohen
- 03.02.2023

## The Data:
- Source: from Kaggle
    - superhero_info.csv:  
    Contains Name, Publisher, Demographic Info, and Body measurements.
    - superhero_powers.csv:  
    Contains Hero name and list of powers

## Tasks
    - Clean the files and combine them into one final DataFrame.
    - Use the combined DataFrame to answer questions.

## Preliminary steps

### Imports

In [1]:
import pandas as pd
import os, json

### Load the data

In [2]:
info_df = pd.read_csv('Data/superhero_info - superhero_info.csv')
display(info_df.head())
display(info_df.info())

Hero|Publisher Gender               Race Alignment  \
0          A-Bomb|Marvel Comics   Male              Human      good   
1  Abe Sapien|Dark Horse Comics   Male      Icthyo Sapien      good   
2            Abin Sur|DC Comics   Male            Ungaran      good   
3     Abomination|Marvel Comics   Male  Human / Radiation       bad   
4   Absorbing Man|Marvel Comics   Male              Human       bad   

  Hair color Eye color Skin color  \
0    No Hair    yellow    Unknown   
1    No Hair      blue       blue   
2    No Hair      blue        red   
3    No Hair     green    Unknown   
4    No Hair      blue    Unknown   

                                   Measurements  
0  {'Height': '203.0 cm', 'Weight': '441.0 kg'}  
1   {'Height': '191.0 cm', 'Weight': '65.0 kg'}  
2   {'Height': '185.0 cm', 'Weight': '90.0 kg'}  
3  {'Height': '203.0 cm', 'Weight': '441.0 kg'}  
4  {'Height': '193.0 cm', 'Weight': '122.0 kg'}

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 463 entries, 0 to 462
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Hero|Publisher  463 non-null    object
 1   Gender          463 non-null    object
 2   Race            463 non-null    object
 3   Alignment       463 non-null    object
 4   Hair color      463 non-null    object
 5   Eye color       463 non-null    object
 6   Skin color      463 non-null    object
 7   Measurements    463 non-null    object
dtypes: object(8)
memory usage: 29.1+ KB


None

In [3]:
powers_df = pd.read_csv('Data/superhero_powers - superhero_powers.csv')
display(powers_df.head())
display(powers_df.info())

hero_names                                             Powers
0      3-D Man         Agility,Super Strength,Stamina,Super Speed
1       A-Bomb  Accelerated Healing,Durability,Longevity,Super...
2   Abe Sapien  Agility,Accelerated Healing,Cold Resistance,Du...
3     Abin Sur                                 Lantern Power Ring
4  Abomination  Accelerated Healing,Intelligence,Super Strengt...

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 667 entries, 0 to 666
Data columns (total 2 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   hero_names  667 non-null    object
 1   Powers      667 non-null    object
dtypes: object(2)
memory usage: 10.5+ KB


None

## Transformations

### Separate Hero|Publisher into 2 columns

In [4]:
# explore the existing format
info_df['Hero|Publisher'].head()

0            A-Bomb|Marvel Comics
1    Abe Sapien|Dark Horse Comics
2              Abin Sur|DC Comics
3       Abomination|Marvel Comics
4     Absorbing Man|Marvel Comics
Name: Hero|Publisher, dtype: object

In [5]:
# separate each string in the column, split them into 2 columns, save the new columns in the dataframe
info_df[['Hero', 'Publisher']] = info_df['Hero|Publisher'].str.split('|',expand=True)
# drop the original column
info_df = info_df.drop(columns=['Hero|Publisher'])
display(info_df.head())

Gender               Race Alignment Hair color Eye color Skin color  \
0   Male              Human      good    No Hair    yellow    Unknown   
1   Male      Icthyo Sapien      good    No Hair      blue       blue   
2   Male            Ungaran      good    No Hair      blue        red   
3   Male  Human / Radiation       bad    No Hair     green    Unknown   
4   Male              Human       bad    No Hair      blue    Unknown   

                                   Measurements           Hero  \
0  {'Height': '203.0 cm', 'Weight': '441.0 kg'}         A-Bomb   
1   {'Height': '191.0 cm', 'Weight': '65.0 kg'}     Abe Sapien   
2   {'Height': '185.0 cm', 'Weight': '90.0 kg'}       Abin Sur   
3  {'Height': '203.0 cm', 'Weight': '441.0 kg'}    Abomination   
4  {'Height': '193.0 cm', 'Weight': '122.0 kg'}  Absorbing Man   

           Publisher  
0      Marvel Comics  
1  Dark Horse Comics  
2          DC Comics  
3      Marvel Comics  
4      Marvel Comics

### Convert Measurements into dictionaries, then separate into 2 columns containing numeric datatypes

In [6]:
# examine a single value from the measurements column
measurements = info_df.loc[0,"Measurements"]
print(type(measurements))
measurements

<class 'str'>


"{'Height': '203.0 cm', 'Weight': '441.0 kg'}"

In [7]:
# use .str.replace to replace all single quotes
info_df['Measurements'] = info_df['Measurements'].str.replace("'",'"')
# use json.loads to load a JSON-compatible STRING into a dictionary/list of dictionaries
info_df['Measurements'] = info_df['Measurements'].apply(json.loads)
info_df['Measurements'].head()

0    {'Height': '203.0 cm', 'Weight': '441.0 kg'}
1     {'Height': '191.0 cm', 'Weight': '65.0 kg'}
2     {'Height': '185.0 cm', 'Weight': '90.0 kg'}
3    {'Height': '203.0 cm', 'Weight': '441.0 kg'}
4    {'Height': '193.0 cm', 'Weight': '122.0 kg'}
Name: Measurements, dtype: object

In [8]:
# use the pandas pd.Series function on the column of dictionaries to automatically convert each key to a new column
height_weight = info_df['Measurements'].apply(pd.Series)
# concatenate the 2 new columns to the original dataframe 
info_df = pd.concat((info_df, height_weight), axis = 1)
# drop the original column
info_df = info_df.drop(columns=['Measurements'])
display(info_df.head())

Gender               Race Alignment Hair color Eye color Skin color  \
0   Male              Human      good    No Hair    yellow    Unknown   
1   Male      Icthyo Sapien      good    No Hair      blue       blue   
2   Male            Ungaran      good    No Hair      blue        red   
3   Male  Human / Radiation       bad    No Hair     green    Unknown   
4   Male              Human       bad    No Hair      blue    Unknown   

            Hero          Publisher    Height    Weight  
0         A-Bomb      Marvel Comics  203.0 cm  441.0 kg  
1     Abe Sapien  Dark Horse Comics  191.0 cm   65.0 kg  
2       Abin Sur          DC Comics  185.0 cm   90.0 kg  
3    Abomination      Marvel Comics  203.0 cm  441.0 kg  
4  Absorbing Man      Marvel Comics  193.0 cm  122.0 kg

In [9]:
# delete string labels in height
info_df['Height'] = info_df['Height'].str.replace(' cm','',regex=False)
# change datatype in height to float
info_df['Height'] = info_df['Height'].astype(float)
# delete string labels in weight
info_df['Weight'] = info_df['Weight'].str.replace(' kg','',regex=False)
# change datatype in weight to float
info_df['Weight'] = info_df['Weight'].astype(float)
display(info_df.head())
display(info_df.info())

Gender               Race Alignment Hair color Eye color Skin color  \
0   Male              Human      good    No Hair    yellow    Unknown   
1   Male      Icthyo Sapien      good    No Hair      blue       blue   
2   Male            Ungaran      good    No Hair      blue        red   
3   Male  Human / Radiation       bad    No Hair     green    Unknown   
4   Male              Human       bad    No Hair      blue    Unknown   

            Hero          Publisher  Height  Weight  
0         A-Bomb      Marvel Comics   203.0   441.0  
1     Abe Sapien  Dark Horse Comics   191.0    65.0  
2       Abin Sur          DC Comics   185.0    90.0  
3    Abomination      Marvel Comics   203.0   441.0  
4  Absorbing Man      Marvel Comics   193.0   122.0

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 463 entries, 0 to 462
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Gender      463 non-null    object 
 1   Race        463 non-null    object 
 2   Alignment   463 non-null    object 
 3   Hair color  463 non-null    object 
 4   Eye color   463 non-null    object 
 5   Skin color  463 non-null    object 
 6   Hero        463 non-null    object 
 7   Publisher   463 non-null    object 
 8   Height      463 non-null    float64
 9   Weight      463 non-null    float64
dtypes: float64(2), object(8)
memory usage: 36.3+ KB


None

###  Convert Powers into one-hot-encoded columns

In [10]:
# examine a single value from the powers column
powers = powers_df.loc[0,"Powers"]
print(type(powers))
powers

<class 'str'>


'Agility,Super Strength,Stamina,Super Speed'

In [11]:
# convert the strings to lists 
powers_df['powers_split'] = powers_df['Powers'].str.split(',')
powers_df['powers_split'].head()

0      [Agility, Super Strength, Stamina, Super Speed]
1    [Accelerated Healing, Durability, Longevity, S...
2    [Agility, Accelerated Healing, Cold Resistance...
3                                 [Lantern Power Ring]
4    [Accelerated Healing, Intelligence, Super Stre...
Name: powers_split, dtype: object

In [12]:
# explode the column of lists
exploded = powers_df.explode('powers_split')
exploded.head()

hero_names                                             Powers  \
0    3-D Man         Agility,Super Strength,Stamina,Super Speed   
0    3-D Man         Agility,Super Strength,Stamina,Super Speed   
0    3-D Man         Agility,Super Strength,Stamina,Super Speed   
0    3-D Man         Agility,Super Strength,Stamina,Super Speed   
1     A-Bomb  Accelerated Healing,Durability,Longevity,Super...   

          powers_split  
0              Agility  
0       Super Strength  
0              Stamina  
0          Super Speed  
1  Accelerated Healing

In [13]:
# drop Nan and save unique values
cols_to_make = exploded['powers_split'].dropna().unique()
cols_to_make

array(['Agility', 'Super Strength', 'Stamina', 'Super Speed',
       'Accelerated Healing', 'Durability', 'Longevity', 'Camouflage',
       'Self-Sustenance', 'Cold Resistance', 'Underwater breathing',
       'Marksmanship', 'Weapons Master', 'Intelligence', 'Telepathy',
       'Immortality', 'Reflexes', 'Enhanced Sight', 'Sub-Mariner',
       'Lantern Power Ring', 'Invulnerability', 'Animation',
       'Super Breath', 'Dimensional Awareness', 'Flight', 'Size Changing',
       'Teleportation', 'Magic', 'Dimensional Travel',
       'Molecular Manipulation', 'Energy Manipulation', 'Power Cosmic',
       'Energy Absorption', 'Elemental Transmogrification',
       'Fire Resistance', 'Natural Armor', 'Heat Resistance',
       'Matter Absorption', 'Regeneration', 'Stealth', 'Power Suit',
       'Energy Blasts', 'Energy Beams', 'Heat Generation', 'Danger Sense',
       'Phasing', 'Force Fields', 'Hypnokinesis', 'Invisibility',
       'Enhanced Senses', 'Jump', 'Shapeshifting', 'Elasticity',
 

In [14]:
# use a for loop and .str.contains to create new columns
for col in cols_to_make:
    powers_df[col] = powers_df['Powers'].str.contains(col)
pd.set_option('display.max_column', None)
powers_df.head()

/var/folders/_5/txdkxz9d07zcw7bs_mc8cnmr0000gn/T/ipykernel_5260/3116508192.py:3: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  powers_df[col] = powers_df['Powers'].str.contains(col)
/var/folders/_5/txdkxz9d07zcw7bs_mc8cnmr0000gn/T/ipykernel_5260/3116508192.py:3: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  powers_df[col] = powers_df['Powers'].str.contains(col)
/var/folders/_5/txdkxz9d07zcw7bs_mc8cnmr0000gn/T/ipykernel_5260/3116508192.py:3: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of callin

hero_names                                             Powers  \
0      3-D Man         Agility,Super Strength,Stamina,Super Speed   
1       A-Bomb  Accelerated Healing,Durability,Longevity,Super...   
2   Abe Sapien  Agility,Accelerated Healing,Cold Resistance,Du...   
3     Abin Sur                                 Lantern Power Ring   
4  Abomination  Accelerated Healing,Intelligence,Super Strengt...   

                                        powers_split  Agility  Super Strength  \
0    [Agility, Super Strength, Stamina, Super Speed]     True            True   
1  [Accelerated Healing, Durability, Longevity, S...    False            True   
2  [Agility, Accelerated Healing, Cold Resistance...     True            True   
3                               [Lantern Power Ring]    False           False   
4  [Accelerated Healing, Intelligence, Super Stre...    False            True   

   Stamina  Super Speed  Accelerated Healing  Durability  Longevity  \
0     True         True                False       False      False   
1     True        False                 True        True       True   
2     True        False                 True        True       True   
3    False        False                False       False      False   
4     True         True                 True       False      False   

   Camouflage  Self-Sustenance  Cold Resistance  Underwater breathing  \
0       False            False            False                 False   
1        True             True            False                 False   
2       False            False             True                  True   
3       False            False            False                 False   
4       False            False            False                 False   

   Marksmanship  Weapons Master  Intelligence  Telepathy  Immortality  \
0         False           False         False      False        False   
1         False           False         False      False        False   
2          True            True          True       True         True   
3         False           False         False      False        False   
4         False           False          True      False        False   

   Reflexes  Enhanced Sight  Sub-Mariner  Lantern Power Ring  Invulnerability  \
0     False           False        False               False            False   
1     False           False        False               False            False   
2      True            True         True               False            False   
3     False           False        False                True            False   
4     False           False        False               False             True   

   Animation  Super Breath  Dimensional Awareness  Flight  Size Changing  \
0      False         False                  False   False          False   
1      False         False                  False   False          False   
2      False         False                  False   False          False   
3      False         False                  False   False          False   
4       True          True                  False   False          False   

   Teleportation  Magic  Dimensional Travel  Molecular Manipulation  \
0          False  False               False                   False   
1          False  False               False                   False   
2          False  False               False                   False   
3          False  False               False                   False   
4          False  False               False                   False   

   Energy Manipulation  Power Cosmic  Energy Absorption  \
0                False         False              False   
1                False         False              False   
2                False         False              False   
3                False         False              False   
4                False         False              False   

   Elemental Transmogrification  Fire Resistance  Natural Armor  \
0   

In [15]:
# drop original Powers column and powers_split column
powers_df = powers_df.drop(columns=['Powers','powers_split'])
powers_df.head()

hero_names  Agility  Super Strength  Stamina  Super Speed  \
0      3-D Man     True            True     True         True   
1       A-Bomb    False            True     True        False   
2   Abe Sapien     True            True     True        False   
3     Abin Sur    False           False    False        False   
4  Abomination    False            True     True         True   

   Accelerated Healing  Durability  Longevity  Camouflage  Self-Sustenance  \
0                False       False      False       False            False   
1                 True        True       True        True             True   
2                 True        True       True       False            False   
3                False       False      False       False            False   
4                 True       False      False       False            False   

   Cold Resistance  Underwater breathing  Marksmanship  Weapons Master  \
0            False                 False         False           False   
1            False                 False         False           False   
2             True                  True          True            True   
3            False                 False         False           False   
4            False                 False         False           False   

   Intelligence  Telepathy  Immortality  Reflexes  Enhanced Sight  \
0         False      False        False     False           False   
1         False      False        False     False           False   
2          True       True         True      True            True   
3         False      False        False     False           False   
4          True      False        False     False           False   

   Sub-Mariner  Lantern Power Ring  Invulnerability  Animation  Super Breath  \
0        False               False            False      False         False   
1        False               False            False      False         False   
2         True               False            False      False         False   
3        False                True            False      False         False   
4        False               False             True       True          True   

   Dimensional Awareness  Flight  Size Changing  Teleportation  Magic  \
0                  False   False          False          False  False   
1                  False   False          False          False  False   
2                  False   False          False          False  False   
3                  False   False          False          False  False   
4                  False   False          False          False  False   

   Dimensional Travel  Molecular Manipulation  Energy Manipulation  \
0               False                   False                False   
1               False                   False                False   
2               False                   False                False   
3               False                   False                False   
4               False                   False                False   

   Power Cosmic  Energy Absorption  Elemental Transmogrification  \
0         False              False                         False   
1         False              False                         False   
2         False              False                         False   
3         False              False                         False   
4         False              False                         False   

   Fire Resistance  Natural Armor  Heat Resistance  Matter Absorption  \
0            False          False            False              False   
1            False          False            False              False   
2            False          False            False              False   
3            False          False            False              False   
4            False          False            False              False   

   Regeneration  Stealth  Power Suit  Energy Blasts  Energy Beams  \
0         False    False  

## Combine the 2 DataFrames into one final DataFrame

In [16]:
# join the DataFrames with pd.merge
heroes = pd.merge(info_df, powers_df, left_on='Hero', right_on='hero_names', how = 'inner')
display(heroes.head())
display(heroes.info())

Gender               Race Alignment Hair color Eye color Skin color  \
0   Male              Human      good    No Hair    yellow    Unknown   
1   Male      Icthyo Sapien      good    No Hair      blue       blue   
2   Male            Ungaran      good    No Hair      blue        red   
3   Male  Human / Radiation       bad    No Hair     green    Unknown   
4   Male              Human       bad    No Hair      blue    Unknown   

            Hero          Publisher  Height  Weight     hero_names  Agility  \
0         A-Bomb      Marvel Comics   203.0   441.0         A-Bomb    False   
1     Abe Sapien  Dark Horse Comics   191.0    65.0     Abe Sapien     True   
2       Abin Sur          DC Comics   185.0    90.0       Abin Sur    False   
3    Abomination      Marvel Comics   203.0   441.0    Abomination    False   
4  Absorbing Man      Marvel Comics   193.0   122.0  Absorbing Man    False   

   Super Strength  Stamina  Super Speed  Accelerated Healing  Durability  \
0            True     True        False                 True        True   
1            True     True        False                 True        True   
2           False    False        False                False       False   
3            True     True         True                 True       False   
4            True    False        False                False        True   

   Longevity  Camouflage  Self-Sustenance  Cold Resistance  \
0       True        True             True            False   
1       True       False            False             True   
2      False       False            False            False   
3      False       False            False            False   
4      False       False            False             True   

   Underwater breathing  Marksmanship  Weapons Master  Intelligence  \
0                 False         False           False         False   
1                  True          True            True          True   
2                 False         False           False         False   
3                 False         False           False          True   
4                 False         False           False         False   

   Telepathy  Immortality  Reflexes  Enhanced Sight  Sub-Mariner  \
0      False        False     False           False        False   
1       True         True      True            True         True   
2      False        False     False           False        False   
3      False        False     False           False        False   
4      False        False     False           False        False   

   Lantern Power Ring  Invulnerability  Animation  Super Breath  \
0               False            False      False         False   
1               False            False      False         False   
2                True            False      False         False   
3               False             True       True          True   
4               False             True      False         False   

   Dimensional Awareness  Flight  Size Changing  Teleportation  Magic  \
0                  False   False          False          False  False   
1                  False   False          False          False  False   
2                  False   False          False          False  False   
3                  False   False          False          False  False   
4                  False   False          False          False  False   

   Dimensional Travel  Molecular Manipulation  Energy Manipulation  \
0               False                   False                False   
1               False                   False                False   
2               False                   False                False   
3               False                   False                False   
4               False                    True                False   

   Power Cosmic  Energy Absorption  Elemental Transmogrification  \
0         False              False                         False   
1         False        

<class 'pandas.core.frame.DataFrame'>
Int64Index: 463 entries, 0 to 462
Columns: 178 entries, Gender to Changing Armor
dtypes: bool(167), float64(2), object(9)
memory usage: 118.9+ KB


None

In [17]:
# move Hero to the first column 
column_to_move = heroes.pop('Hero')
heroes.insert(0, 'Hero', column_to_move)
# delete hero_names column
heroes = heroes.drop(columns=['hero_names'])
display(heroes.head())

/var/folders/_5/txdkxz9d07zcw7bs_mc8cnmr0000gn/T/ipykernel_5260/3210628752.py:3: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  heroes.insert(0, 'Hero', column_to_move)


Hero Gender               Race Alignment Hair color Eye color  \
0         A-Bomb   Male              Human      good    No Hair    yellow   
1     Abe Sapien   Male      Icthyo Sapien      good    No Hair      blue   
2       Abin Sur   Male            Ungaran      good    No Hair      blue   
3    Abomination   Male  Human / Radiation       bad    No Hair     green   
4  Absorbing Man   Male              Human       bad    No Hair      blue   

  Skin color          Publisher  Height  Weight  Agility  Super Strength  \
0    Unknown      Marvel Comics   203.0   441.0    False            True   
1       blue  Dark Horse Comics   191.0    65.0     True            True   
2        red          DC Comics   185.0    90.0    False           False   
3    Unknown      Marvel Comics   203.0   441.0    False            True   
4    Unknown      Marvel Comics   193.0   122.0    False            True   

   Stamina  Super Speed  Accelerated Healing  Durability  Longevity  \
0     True        False                 True        True       True   
1     True        False                 True        True       True   
2    False        False                False       False      False   
3     True         True                 True       False      False   
4    False        False                False        True      False   

   Camouflage  Self-Sustenance  Cold Resistance  Underwater breathing  \
0        True             True            False                 False   
1       False            False             True                  True   
2       False            False            False                 False   
3       False            False            False                 False   
4       False            False             True                 False   

   Marksmanship  Weapons Master  Intelligence  Telepathy  Immortality  \
0         False           False         False      False        False   
1          True            True          True       True         True   
2         False           False         False      False        False   
3         False           False          True      False        False   
4         False           False         False      False        False   

   Reflexes  Enhanced Sight  Sub-Mariner  Lantern Power Ring  Invulnerability  \
0     False           False        False               False            False   
1      True            True         True               False            False   
2     False           False        False                True            False   
3     False           False        False               False             True   
4     False           False        False               False             True   

   Animation  Super Breath  Dimensional Awareness  Flight  Size Changing  \
0      False         False                  False   False          False   
1      False         False                  False   False          False   
2      False         False                  False   False          False   
3       True          True                  False   False          False   
4      False         False                  False   False          False   

   Teleportation  Magic  Dimensional Travel  Molecular Manipulation  \
0          False  False               False                   False   
1          False  False               False                   False   
2          False  False               False                   False   
3          False  False               False                   False   
4          False  False               False                    True   

   Energy Manipulation  Power Cosmic  Energy Absorption  \
0                False         False              False   
1                False         False              False   
2                False         False              False   
3                False         False              False   
4                False         False               True   

   Elemental Transmogrification  Fire Resistance  Natural Arm

## Compare the average weight of super heroes who have Super Speed to those who do not.

In [18]:
mean_weight_vs_Super_Speed = heroes.groupby('Super Speed')['Weight'].mean()
mean_weight_vs_Super_Speed

Super Speed
False    101.773585
True     129.404040
Name: Weight, dtype: float64

- The average weight for Heroes with Super Speed is 129.40 kg.
- The average weight for Heroes without Super Speed is 101.77 kg.
- Heroes with Super Speed are 27.63 kg heavier (on average) than Heroes without Super Speed.

## What is the average height of heroes for each publisher?

In [19]:
mean_height_vs_publisher = heroes.groupby('Publisher')['Height'].mean()
mean_height_vs_publisher

Publisher
DC Comics            181.923913
Dark Horse Comics    176.909091
George Lucas         159.600000
Image Comics         211.000000
Marvel Comics        191.546128
Shueisha             171.500000
Star Trek            181.500000
Team Epic TV         180.750000
Unknown              178.000000
Name: Height, dtype: float64

- The average height for George Lucas Heroes is 159.60 cm.
- The average height for Shueisha Heroes is 171.50 cm.
- The average height for Dark Horse Comics Heroes is 176.91 cm.
- The average height for Unknown Heroes is 178.00 cm.
- The average height for Team Epic TV Heroes is 180.75 cm.
- The average height for Star Trek Heroes is 181.5 cm.
- The average height for DC Comics Heroes is 181.92 cm.
- The average height for Marvel Comics Heroes is 191.55 cm.
- The average height for Image Comics Heroes is 211.00 cm.